app/views/ProfileView/index.js

```diff
@@ -361,6 +378,73 @@ class ProfileView extends React.Component {
 		);
 	}
 
+	renderNicknameField = () => {
+		const {
+			nickname
+		} = this.state;
+		const {
+			theme, serverVersion
+		} = this.props;
+
+		// Nickname was added to release 3.5.0
+		if (semver.lte(serverVersion, '3.5.0')) {
+			return null;
+		}
+
+		return (
+			<RCTextInput
+				editable
+				inputStyle={[
+					!true && styles.disabled
+				]}
+				maxLength={120}
+				inputRef={(e) => { this.nickname = e; }}
+				label={I18n.t('Nickname')}
+				placeholder={I18n.t('Nickname')}
+				value={nickname}
+				onChangeText={value => this.setState({ nickname: value })}
+				onSubmitEditing={() => { this.bio.focus(); }}
+				testID='profile-view-nickName'
+				theme={theme}
+			/>
+		);
+	}
+
+	renderBioField = () => {
+		const {
+			bio
+		} = this.state;
+		const {
+			theme, serverVersion
+		} = this.props;
+
+		// Bio was added to release 3.5.0
+		if (semver.lte(serverVersion, '3.1.0')) {
+			return null;
+		}
+
+		return (
+			<RCTextInput
+				editable
+				inputStyle={[
+					!true && styles.disabled,
+					styles.bio
+				]}
+				numberOfLines={2}
+				maxLength={260}
```

Comment:
Where did you get this from?

Reply:
https://github.com/RocketChat/Rocket.Chat/blob/10ea922d89b05f358c656996913d6492edd4c417/server/methods/saveUserProfile.js#L50